### PR TITLE
#498. Implementend+tested Resignations#register.

### PR DESCRIPTION
**Pull request by criske**
PR for #498

### CODE REVIEW

**Comment by criske** [comment]
@amihaiemil Wouldn't be better if `Tasks#unassign(Task)` call internally `Resignations#register` too? Now `Tasks#unassign(Task, Reason)` would return `Resignation` instead of just `Task`.

File: self-api/src/main/java/com/selfxdsd/api/Task.java [hunk 1]
```diff
@@ -30,6 +30,9 @@
  * @author Mihai Andronache (amihaiemil@gmail.com)
  * @version $Id$
  * @since 0.0.1
+ * @todo #502:30min. Task api should expose resignations(), for easier access
```
Comment (amihaiemil) [inline self-api/src/main/java/com/selfxdsd/api/Task.java:33]: @criske 498 is the parent ticket here, not 502

**Review by amihaiemil — COMMENTED** [review]
@criske just one tiny comment

**Comment by amihaiemil** [comment]
@criske 

> Wouldn't be better if Tasks#unassign(Task) call internally Resignations#register too

I was also thinking about that option, but I'm not sure. I think maybe it's more clear if we keep them separate.
Plus, resignation should not always be registered -- for example, if the Task is deregistered (removed) by the PO or ARC, then we should not register a resignation.

**Comment by criske** [comment]
@amihaiemil 
Calling  `Resignations#register` and `Tasks#unassign(Task)` sequentially might create problems imo  -- I think this in the order when execting a possible ResignStep.
These two should be executed in a transaction at the db level (supposing there are two tables `tasks` and `resignation_tasks`).

**Comment by amihaiemil** [comment]
@criske yeah, they should theoretically be executed in the same transaction. But I think it's a bit overkill. It's not a big problem if registering the resignation fails.

So Task.unassign + Resignations.register, sequencially in this order should be fine... at least for now.

**Comment by amihaiemil** [comment]
@rultor merge it please

**Pull request merged**